### DISCUSSION TITLE
空码的处理问题

### DISCUSSION THREAD

**Discussion by selong1234** [amorphobia/openfly#47] [问与答/Q&A]
大佬，你好，这个版本能不能实现小鹤原版中关于空码检索次显的设置，就是次显的编码还是原来的，只不过如果前面是空码的话，就显示次显。开源小鹤现在是直接把空码补上了。
比如：阿爸 小鹤音形里的编码是aaba，但是aab是空码，所以打aab会出现阿爸a，但阿爸的编码还是aaba。
开源小鹤是直接aab的编码给了阿爸a。这样做会出现的问题是，如果自己加一个aab编码的词，小鹤音形中，输入aab，就直接出现自己造的词，而开源小鹤则会出现重码，包括阿爸a和自己造的词。
因为不懂lua，所以不知道程序中是怎么处理这些空码的，麻烦大佬看看开源小鹤能不能实现上面的功能。
看到之前的讨论了，我先研究一下你之前发的东西

**Comment by amorphobia** [comment]
这里实现的方法是单独用一个词典存了这些空码提示 (https://github.com/amorphobia/openfly/blob/main/openfly.embedded.hint.dict.yaml)

本来在设定里打开编码提示也能实现空码提示，但是提示词的顺序只能按照编码的字母序，因此小鹤的作者曾经提过这个 issue: https://github.com/rime/librime/issues/297  想让提示顺序按照词典顺序，结果是无法满足的，所以打开编码提示的意义也不大。

回到你的问题本身，openfly 有一个 lua 脚本可以实现你的需求，你可以在自定义词典里加上如下一条：
阿爸`[删]	aab

这样就可以删除 aab 上的空码提示词“阿爸”。

**Reply by selong1234** [reply]
大佬 我在你之前写的那个讨论里面回复了 麻烦你再看一下

**Reply by amorphobia** [reply]
那个方案应该不是很成熟，建议你采用这里删词的方法

**Reply by selong1234** [reply]
删词这个感觉没有之前你写的那个方便，加一个词然后还得删一个词，可能那个方法有些问题我没有发现，暂时就看到我提的那两个问题。
然后还想请教一下，你这里也提到编码提示，这个编码提示在哪里能控制开关啊，没找到地方，还请大佬给提示一下。

**Reply by amorphobia** [reply]
https://rime-aca.tumblr.com/post/67241713724/rime%E6%96%B9%E6%A1%88%E8%A3%BD%E4%BD%9C%E8%A9%B3%E8%A7%A3

我也是刚搜的，应该是这里提到的 `enable_completion`

**Reply by selong1234** [reply]
enable_completion: 提前顯示尚未輸入完整碼的字〔僅table_translator有效〕
大佬  这不是你说的那个自动不全吗 不是编码提示啊

**Reply by amorphobia** [reply]
对不起，我一直想的就是自动补全，说成编码提示了

**Reply by selong1234** [reply]
我刚才试了一下，如果单独把这个打开，不加你之前写的那个空码的lua程序，那就是编码提示功能啊，打一个字母出来一堆候选。。。

**Reply by amorphobia** [reply]
是的，编码提示的顺序是编码的字母顺序，无法实现小鹤官方的词库顺序，如果希望按照词库顺序，我想到的只有两种方案

1. [关闭编码提示] 把首选编码提示的词单独放一个词典（目前方案），遇到自定义词语重复的就删词
2. [打开编码提示] 用 https://github.com/amorphobia/openfly/discussions/33 这里的脚本，并且给词库里所有词条加上权重

第二种方案工作量太大了，所以不建议

**Reply by selong1234** [reply]
大佬，没明白你说的字母顺序和词库顺序的区别，能举个例子吗，我现在用你的第二种方案，没感觉顺序有什么问题呢

**Reply by selong1234** [reply]
呃 刚才试了几个别的词 发现问题了

**Reply by selong1234** [reply]
感觉它也不是字母顺序，比如打aam，出来的是阿妹~w，不是阿妈

**Reply by selong1234** [reply]
小鹤打aam，出的也是阿妹

**Reply by amorphobia** [reply]
我也忘了是什么顺序了，总之和小鹤的顺序有不一样的

**Reply by selong1234** [reply]
哦，我再试试，不行我就用你说的第一种方法，谢谢大佬

**Reply by selong1234** [reply]
大佬，这个是不是librime改过了，我刚才试了100个词，都是和你加的那个词典是一样的

**Reply by selong1234** [reply]
是我搞错了，应该是我之前部署的时候生成的bin码表里面，还有你加的词典，所以才按你的顺序来的，我重新生成一个新的bin码表，顺序就不一样了，是按字母顺序了。
但是这样也说明了，如果生成码表时就把你的字典加进去，然后再用你的代码，就不用手动排序了，应该会是按你加的字典里面的顺序。
